src/Service/AssetFilterManager.php

```diff
@@ -63,49 +58,84 @@ protected function setConfig($config)
     /**
      * See if there are filters for the asset, and if so, set them.
      *
-     * @param   string          $path
-     * @param   AssetInterface  $asset
+     * @param string         $path
+     * @param AssetInterface $asset
      *
      * @throws Exception\RuntimeException on invalid filters
      */
     public function setFilters($path, AssetInterface $asset)
     {
+        $filters = $this->getFilters($path, $asset);
+
+        foreach ($filters as $filter) {
+            $this->setFilter($filter, $asset);
+        }
+    }
+
+    /**
+     * Get the filters from config based on path, mimetype or extension.
+     *
+     * @param string $path
+     * @param AssetInterface $asset
+     *
+     * @return array|mixed
+     */
+    protected function getFilters($path, AssetInterface $asset)
+    {
         $config = $this->getConfig();
 
         if (!empty($config[$path])) {
-            $filters = $config[$path];
-        } elseif (!empty($config[$asset->mimetype])) {
-            $filters = $config[$asset->mimetype];
-        } else {
-            $extension = $this->getMimeResolver()->getExtension($asset->mimetype);
-            if (!empty($config[$extension])) {
-                $filters = $config[$extension];
-            } else {
-                return;
-            }
+            return $config[$path];
         }
 
-        foreach ($filters as $filter) {
-            if (is_null($filter)) {
-                continue;
-            }
-            if (!empty($filter['filter'])) {
-                $this->ensureByFilter($asset, $filter['filter']);
-            } elseif (!empty($filter['service'])) {
-                $this->ensureByService($asset, $filter['service']);
-            } else {
-                throw new Exception\RuntimeException(
-                    'Invalid filter supplied. Expected Filter or Service.'
-                );
-            }
+        if (!empty($asset->mimetype) && !empty($config[$asset->mimetype])) {
+            return $config[$asset->mimetype];
         }
+
+        $extension = strtolower(pathinfo($asset->getSourcePath(), PATHINFO_EXTENSION));
+
+        if (!empty($config[$extension])) {
+            return $config[$extension];
+        }
+
+        return [];
+    }
+
+    /**
+     * Set the filter by filter or service
+     *
+     * @param string         $filter
+     * @param AssetInterface $asset
+     */
+    protected function setFilter($filter, AssetInterface $asset)
+    {
+        if (is_null($filter)) {
+            return;
```

Comment:
It feels like this should be an error. What is the use case for this "failing" silently?

Reply:
I see what you mean. However, this just keeps it's current behavior. All I did was de-nest all the if statements.

Reply:
After reading the code... it seems this is here if you set the filters to null in your config (which you would do if you overwrite the global config with local config)...  Like so:

```php
return [
    'AssetManager' => [
        'filters' => [
            'css' => [
                'filter_one' => null
             ],
        ]
    ];
];
```

Everything still works.  If we error out here, we'll have no way to stop using a filter on say a dev box, without replacing all the filters, which may not be what we really want.